graylog2-web-interface/src/views/components/visualizations/__tests__/ChartData.test.js

```diff
@@ -86,6 +86,28 @@ describe('Chart helper functions', () => {
       expect(result).toHaveLength(6);
       expect(result).toEqual(expectedResult);
     });
+    it('should allow passing a format series function to modify the series structure', () => {
+      const input = readFixture('ChartData.test.oneColumOneRowPivot.json');
+      const generatorFunction = (type, name, x, y, z) => ({ type, name, x, y, z });
+      const formatSeriesCustom = ({ valuesBySeries, xLabels }) => {
+        // In this example we want to create only one series, with an z value, which contains all series data
+        const z: Array<any> = Object.values(valuesBySeries).map((series) => {
+          const newSeries = fill(Array(xLabels.length), null);
+          return merge(newSeries, series);
+        });
```

Comment:
By default the function would return nothing for missing values in the `valuesBySeries` arrays. This is not problem when working with plotly, but it's a problem when saving the result as JSON. If just added nulls as a workaround